pkg/server/plugin/credentialcomposer/v1_test.go

```diff
@@ -0,0 +1,941 @@
+		})
+	}
+}
+
+func TestV1ComposeWorkloadJWTSVID(t *testing.T) {
+	id := spiffeid.RequireFromString("spiffe://domain.test/workload")
+	for _, tt := range []struct {
+		test      string
+		pluginErr error
+
+		idIn            spiffeid.ID
+		attributesIn    credentialcomposer.JWTSVIDAttributes
+		expectRequestIn *credentialcomposerv1.ComposeWorkloadJWTSVIDRequest
+
+		responseOut         *credentialcomposerv1.ComposeWorkloadJWTSVIDResponse
+		expectAttributesOut credentialcomposer.JWTSVIDAttributes
+
+		expectCode    codes.Code
+		expectMessage string
+	}{
+		{
+			test:          "invalid ID",
+			expectCode:    codes.Internal,
+			expectMessage: "credentialcomposer(test): invalid workload ID: empty",
+		},
+		{
+			test:          "plugin fails",
+			idIn:          id,
+			attributesIn:  credentialcomposer.JWTSVIDAttributes{Claims: map[string]interface{}{"ORIGINAL_KEY": "ORIGINAL_VALUE"}},
+			pluginErr:     status.Error(codes.Internal, "oh no"),
+			expectCode:    codes.Internal,
+			expectMessage: "credentialcomposer(test): oh no",
+		},
+		{
+			test:          "invalid claims input",
+			idIn:          id,
+			attributesIn:  credentialcomposer.JWTSVIDAttributes{},
+			expectCode:    codes.Internal,
+			expectMessage: "credentialcomposer(test): invalid workload JWTSVID attributes: invalid claims: cannot be empty",
+		},
+		{
+			test:         "attributes unchanged if unimplemented",
+			pluginErr:    status.Error(codes.Unimplemented, "not implemented"),
+			idIn:         id,
+			attributesIn: credentialcomposer.JWTSVIDAttributes{Claims: map[string]interface{}{"ORIGINAL_KEY": "ORIGINAL_VALUE"}},
+			expectRequestIn: &credentialcomposerv1.ComposeWorkloadJWTSVIDRequest{
+				SpiffeId: id.String(),
+				Attributes: &credentialcomposerv1.JWTSVIDAttributes{
+					Claims: &structpb.Struct{Fields: map[string]*structpb.Value{"ORIGINAL_KEY": structpb.NewStringValue("ORIGINAL_VALUE")}},
+				},
+			},
+			responseOut:         &credentialcomposerv1.ComposeWorkloadJWTSVIDResponse{},
+			expectAttributesOut: credentialcomposer.JWTSVIDAttributes{Claims: map[string]interface{}{"ORIGINAL_KEY": "ORIGINAL_VALUE"}},
+		},
+		{
+			test:         "attributes unchanged if plugin does not respond with attributes",
+			idIn:         id,
+			attributesIn: credentialcomposer.JWTSVIDAttributes{Claims: map[string]interface{}{"ORIGINAL_KEY": "ORIGINAL_VALUE"}},
+			expectRequestIn: &credentialcomposerv1.ComposeWorkloadJWTSVIDRequest{
+				SpiffeId: id.String(),
+				Attributes: &credentialcomposerv1.JWTSVIDAttributes{
+					Claims: &structpb.Struct{Fields: map[string]*structpb.Value{"ORIGINAL_KEY": structpb.NewStringValue("ORIGINAL_VALUE")}},
+				},
+			},
+			responseOut:         &credentialcomposerv1.ComposeWorkloadJWTSVIDResponse{},
+			expectAttributesOut: credentialcomposer.JWTSVIDAttributes{Claims: map[string]interface{}{"ORIGINAL_KEY": "ORIGINAL_VALUE"}},
+		},
+		{
+			test:         "attributes unchanged if plugin does not respond with attributes",
```

Comment:
Test description needs to be updated here.

Reply:
:+1: